packages/client/test/test-utils/customMatchers.ts

```diff
@@ -0,0 +1,61 @@
+import { expect } from '@jest/globals'
+import type { MatcherState } from 'expect'
+import { printExpected, printReceived } from 'jest-matcher-utils'
+import { isFunction } from 'lodash'
+import { StreamrClientError, StreamrClientErrorCode } from './../../src/StreamrClientError'
+
+// we could ES2015 module syntax (https://jestjs.io/docs/expect#expectextendmatchers),
+// but the IDE doesn't find custom matchers if we do that
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace jest {
+        interface Matchers<R> {
+            toThrowStreamError(expectedError: { code: StreamrClientErrorCode, message?: string }): R
+        }
+    }
+}
+
+const formError = (description: string, expected: string, actual: string) => {
+    return {
+        pass: false,
+        message: () => `${description}\nExpected: ${printExpected(expected)}\nReceived: ${printReceived(actual)}`
+    }
+}
+
+const toThrowStreamError = function(
+    this: MatcherState,
+    actual: unknown, // should be (() => StreamrClientError) | StreamrClientError
+    expectedError: { code: StreamrClientErrorCode, message?: string }
+) {
+    let actualError
+    if (isFunction(actual)) {
+        try {
+            actual()
+            return {
+                pass: false,
+                message: () => `Function didn't throw`
+            }
+        } catch (e) {
+            actualError = e
+        }
+    } else {
+        actualError = actual
+    }
+    if (!(actualError instanceof StreamrClientError)) {
```

Comment:
Line break before to separate logically.
```suggestion
    if (!(actualError instanceof StreamrClientError)) {
```